### PR TITLE
Add unit test for bucket merge with partially filled diff

### PR DESCRIPTION
**Pull request by toregge**
@vekterli : please review
@geirst , @baldersheim : FYI

### CODE REVIEW

File: storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp [hunk 1]
```diff
@@ -21,12 +80,14 @@ MergeStatus::~MergeStatus() = default;
 bool
 MergeStatus::removeFromDiff(
         const std::vector<api::ApplyBucketDiffCommand::Entry>& part,
-        uint16_t hasMask)
+        uint16_t hasMask,
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp:83]: Consider adding a function comment that clarifies that this is the _per-request_ mask, not the global mask (and maybe also why that is why we need to additionally provide the original node set)

File: storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp [hunk 2]
```diff
@@ -62,11 +123,14 @@ MergeStatus::removeFromDiff(
                 }
                 it = diff.erase(it);
                 altered = true;
-            } else if (it2->_entry._hasMask != it->_hasMask) {
-                // Hasmasks have changed, meaning bucket contents changed on
-                // one or more of the nodes during merging.
-                altered = true;
-                it->_hasMask = it2->_entry._hasMask;
+            } else {
+                uint16_t mask = remap_mask(it2->_entry._hasMask);
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp:127]: Consider adding a comment on why we have to remap the masks here

**Review by vekterli — APPROVED** [review]
:+1: just have a couple of requests for additional comments that will make it easier to remember the hows and whys a few years down the line 🙂

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp [hunk 3]
```diff
@@ -3,11 +3,70 @@
...
 
 namespace storage {
 
+namespace {
+
+/*
+ * Class for remapping bit masks from a partial set of nodes to a full
+ * set of nodes.
+ */
+class MaskRemapper
```
Comment (geirst) [inline storage/src/vespa/storage/persistence/filestorage/mergestatus.cpp:19]: @vekterli and @toregge : Consider moving this out of the cpp file and then creating explicit unit tests for it.

**Pull request merged**